kyo-core/shared/src/main/scala/kyo/AllowUnsafe.scala

```diff
@@ -0,0 +1,28 @@
+package kyo
+
+import kyo.Ansi.*
+import kyo.kernel.Safepoint
+import scala.annotation.implicitNotFound
+
+@implicitNotFound("""
+Unsafe operation detected!
+This code requires explicit permission to perform unsafe operations that
+break referential transparency.
+
+Recommended approach:
+Wrap your unsafe code in IO(...) to properly manage side effects:
+
+IO { /* your unsafe code here */ }
```

Comment:
I'm having second thoughts on `IO.apply` providing the `AllowUnsafe` evidence. It'd encourage the use of unsafe APIs that are meant only for more advanced users for performance or integrations. I'm considering adding a new `IO.Unsafe.apply` method instead. We could also move `IO.run` to `IO.Unsafe.run`. Wdyt?

@steinybot I think you're leaning toward being more relaxed in relation to referential transparency but that's an essential aspect to ensure Kyo's effects compose correctly. Have you considered using the direct syntax? It drops referential transparency but keeps composability since it gets translated to referentially-transparent code.

Reply:
I've introduced `IO.Unsafe.apply`, which is similar to `IO.apply` but provides the `AllowUnsafe` evidence, and moved the IO run methods to it.